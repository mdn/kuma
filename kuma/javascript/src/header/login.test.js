//@flow
import React from 'react';
import { create } from 'react-test-renderer';
import Dropdown from './dropdown.jsx';
import Login from './login.jsx';
import UserProvider from '../user-provider.jsx';

test('Login snapshot before user data is fetched', () => {
    const login = create(
        <UserProvider.context.Provider value={null}>
            <Login />
        </UserProvider.context.Provider>
    ).toJSON();
    expect(login).toBe(null);
});

test('Login component when user is not logged in', () => {
    const login = create(
        <UserProvider.context.Provider value={UserProvider.defaultUserData}>
            <Login />
        </UserProvider.context.Provider>
    ).toJSON();
    expect(login.children[0]).toBe('Sign in');
    expect(login).toMatchSnapshot();
});

test('Login component when user is logged in', () => {
    const login = create(
        <UserProvider.context.Provider
            value={{
                ...UserProvider.defaultUserData,
                isAuthenticated: true,
                username: 'test-username',
                avatarUrl: 'test-url'
            }}
        >
            <Login />
        </UserProvider.context.Provider>
    );

    expect(login.toJSON()).toMatchSnapshot();

    // Expect a Dropdown element
    let root = login.root;
    let dropdown = root.findByType(Dropdown);
    expect(dropdown).toBeDefined();

    // Whose label prop is an image with the expected src and alt attributes
    expect(dropdown.props.label.props.srcSet).toContain('test-url 50w');
    expect(dropdown.props.label.props.srcSet).toContain('test-url 200w');
    expect(dropdown.props.label.props.alt).toEqual('test-username');

    let string = JSON.stringify(login.toJSON());
    expect(string).toContain('View profile');
    expect(string).toContain('Edit profile');
    expect(string).toContain('Sign out');
});
