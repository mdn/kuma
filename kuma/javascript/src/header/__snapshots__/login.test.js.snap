// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component when user is logged in 1`] = `
<div
  className="auth-container"
>
  <div
    className="dropdown-container"
  >
    <button
      aria-haspopup="true"
      className="dropdown-menu-label"
      type="button"
    >
      <img
        alt="test-username"
        className="avatar"
        src="/static/img/avatar.png"
        srcSet="test-url 200w, test-url 50w"
      />
    </button>
    <ul
      className="dropdown-menu-items"
      role="menu"
      style={
        Object {
          "right": 0,
        }
      }
    >
      <li>
        <a
          href="/en-US/profiles/test-username"
        >
          View profile
        </a>
      </li>
      <li>
        <a
          href="/en-US/profiles/test-username/edit"
        >
          Edit profile
        </a>
      </li>
      <li>
        <form
          action="/en-US/users/signout"
          method="post"
        >
          <input
            name="next"
            type="hidden"
            value="/"
          />
          <button
            className="signout-button"
            type="submit"
          >
            Sign out
          </button>
        </form>
      </li>
    </ul>
  </div>
</div>
`;

exports[`Login component when user is not logged in 1`] = `
<a
  className="signin-link"
  data-service="GitHub"
  href="/users/account/signup-landing?next=/"
  rel="nofollow"
>
  Sign in
</a>
`;
